torch/csrc/distributed/c10d/ProcessGroupMPI.cpp

```diff
@@ -267,8 +267,8 @@ c10::intrusive_ptr<ProcessGroupMPI> ProcessGroupMPI::createProcessGroupMPI(
 
     // If no ranks are specified, assume we're creating the root group
     if (!ranks.empty()) {
-      MPI_Group worldGroup;
-      MPI_Group ranksGroup;
+      MPI_Group worldGroup = nullptr;
+      MPI_Group ranksGroup = nullptr;
```

Comment:
Hi, @cyyever.
I got a compile error here with gcc11
```
/localdisk/haozhe/pt-upstream/pytorch/torch/csrc/distributed/c10d/ProcessGroupMPI.cpp:270:30: error: cannot convert ‘std::nullptr_t’ to ‘MPI_Group’ {aka ‘int’} in initialization
  270 |       MPI_Group worldGroup = nullptr;
```
Should it be changed to:
`MPI_Group worldGroup = 0;` or some other init value?

Reply:
@zhuhaozhe  Oh, I will fix them.